crates/store/re_dataframe/src/query.rs

```diff
@@ -1230,7 +1226,21 @@ impl<E: StorageEngineLike> QueryHandle<E> {
     #[inline]
     pub fn next_row_batch(&self) -> Option<ArrowRecordBatch> {
         let row = self.next_row()?;
-        ArrowRecordBatch::try_new(self.schema().clone(), row).ok()
+        match ArrowRecordBatch::try_new_with_options(
+            self.schema().clone(),
+            row,
+            &arrow::array::RecordBatchOptions::new().with_row_count(Some(1)),
```

Comment:
Explicitly setting row-count to one means it works even when there are no columns (e.g. due to heavy filtering)

Reply:
This comment should be in the code..!

Reply:
Right you are!